haystack/pipelines/base.py

```diff
@@ -405,14 +405,19 @@ def eval(
             if params is None:
                 params = {}
             params["add_isolated_node_eval"] = True
-        queries = [label.query for label in labels]
-        for query, label in zip(queries, labels):
-            predictions = self.run(query=query, labels=label, params=params, debug=True)
+        for label in labels:
+            if label.filters is not None:
+                if params is None:
+                    params = {"filters": label.filters}
+                else:
+                    # join both filters and overwrite filters in params with filters in labels 
+                    params["filters"] = {**params.get("filters", {}), **label.filters}
+            predictions = self.run(query=label.query, labels=label, params=params, debug=True)
```

Comment:
`labels=label` looks weird here, although it was like that before.

Reply:
I agree that it looks weird but I still wouldn't rename it. The `run` method's parameter is called `labels` although it's a single `MultiLabel` now, whereas the `eval` method has a `labels` parameter that is a List[MultiLabel]. Once we have a `run_batch` method it will look better.
What's your suggestion? Renaming `label` within this method to `multilabel`? I wouldn't prefer encoding the type in the name.

Reply:
`labels_per_query_id` would match best if we added `query_id` to `MultiLabel`. Improving that when `run_batch` is there would be ok.